### ISSUE TITLE
array-scalar dispatch in ^

### ISSUE THREAD

**Issue by goldingn** [greta-dev/greta#298]
Raising a greta array with more than two dimensions (and more than one element) to the power of a scalar warns on definition and errors on sampling.

Reprex:
```r
library(greta)
x <- normal(0, 1, dim = c(2, 1, 1))
y <- x ^ 2
```
> ```
> Warning messages:
> 1: In (function (..., na.rm = FALSE)  :
>   an argument will be fractionally recycled
> 2: In (function (..., na.rm = FALSE)  :
>   an argument will be fractionally recycled
> ```
```r
m <- model(y)
draws <- mcmc(m)
```
> ```
> Error in py_call_impl(callable, dots$args, dots$keywords) : 
>   InvalidArgumentError: Incompatible shapes: [4,2,1,1] vs. [4,1,1]
> [[node Pow (defined at /util/dispatch.py:180) ]]
> ...
> ```

greta version 0.3.0.9008

**Comment by goldingn** [comment]
This is because of a mismatch in the number of dimensions in the corresponding tensorflow code (3 vs 2), and because Tensorflow's broadcasting of scalars is apparently broken by forcing the corresponding tensors to be 2D. It's unclear how many other functions and operators this affects.

The quickest fix would be to internally check for scalars in these functions and pad their dimensions as necessary. This could be done in some of the templating functions that already exist to extend the batch dimension of greta arrays.

Tests for this behaviour should also be added to the test suite, possibly within `check_op` by extending one of the greta arrays with an extra dimension.

**Comment by goldingn** [comment]
Closed by #329
